### PR TITLE
Add NoUnneededCurlyBracesFixer

### LINKED ISSUE #1540: Feature request: `case` without curly braces
To avoid something like this:

``` php
switch ($foo) {
    case 'bar': {
        // ...
    }
}
```

### LINKED ISSUE #2798: no_unneeded_control_parentheses have no effect
PHP version `5.6.30`, CS Fixer version `2.3.1` 

I used command  `./vendor/bin/php-cs-fixer fix -v --diff --dry-run <file path>`

I have set `'no_unneeded_control_parentheses' => ['break', 'clone', 'continue', 'echo_print', 'return', 'switch_case', 'yield'],` in my configuration. 

Here is the sample code: 

```
switch ($status) {
    case 2: {
       // Single line 
    }
    break;

    case 3: {
       // Multline code 
       // Multline code 
       // Multline code 
    }
    break;

}
```

CS Fixer don't detect any error here, as parentheses are not needed at least in one line code block. 

Can any one help out in it.

### PR DESCRIPTION
**Pull request by SpacePossum**
replaces https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2799

closes #2798
closes #1540

$ php php-cs-fixer describe no_unneeded_curly_braces
Description of no_unneeded_curly_braces rule.
Removes unneeded curly braces.

Fixing examples:
 * Example #1.
```diff
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ @@
   -<?php { echo 1; }
   +<?php  echo 1;
   ----------- end diff -----------
```

tested on this repo and SF

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
For the record, 👎 on this fixer existing.

**Comment by GrahamCampbell** [comment]
Oh, wait. It doesn't do what I thought... maybe it needs renaming?

**Comment by julienfalque** [comment]
@GrahamCampbell What confused you exactly? The fixer's name looks good to me.

**Comment by keradus** [comment]
as he didn't provided any details, i bet he was thinking about braces in:
```php
if (true) {
    echo "foo";
}
```

which is not needed in fact, but we want to keep it (and this fixer is not removing them).

more precise name, yet more ugly, would be sth like `NoBracesOfStatementGroupWithoutControlStructureFixer`

**Comment by SpacePossum** [comment]
>NoBracesOfStatementGroupWithoutControlStructureFixer

how about;

>NoCurlyBracesWithoutControlStructureFixer

?  not sure about the namespace of the fixer though

**Comment by julienfalque** [comment]
I'm fine with current name, maybe it's better to make the description more explicit about what is an _unneeded curly brace_ instead? Like _Removes curly braces that are superfluous and don't wrap a control structure's body._ WDYT?

**Comment by SpacePossum** [comment]
Can we please remove fabbot.io as service? It provides no meaning as we tests the CS of the code better.
I know it is some way to see what version of the fixer fabbot.io uses. However that is not of interest of most contributors and all are effected by the failed check as PR's show as failed, both in on the person PR over as well in the PR overview of the repo.

**Comment by keradus** [comment]
I would rather detect that our cofig is too future-aware and skip fabbot in given situation. Yet, nothing this PR is related to.

**Comment by SpacePossum** [comment]
Can we than remove fabbot.io as being required to pass?
It is related to this PR, as I feel not acting on fail status of this PR because I assume false negative of fabbot.io while there might be a legit failure of travis.

**Comment by keradus** [comment]
> Can we than remove fabbot.io as being required to pass?

it is not required, required are only AppVeyor and Travis
your concern is about github ui directly, as they display overall status on top of all integrations, not only required ones.

**Comment by keradus** [comment]
done: #2935

File: tests/Fixer/ControlStructure/NoUnneededCurlyBracesFixerTest.php [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    public function testNoFix7($expected)
+    {
+        $this->doTest($expected);
+    }
+
+    public function provideNoFixCases7()
+    {
+        return [
+            [
+                '<?php
```
Comment (julienfalque) [inline tests/Fixer/ControlStructure/NoUnneededCurlyBracesFixerTest.php]: Why do the two first test cases here require PHP 7?

Comment (SpacePossum) [inline tests/Fixer/ControlStructure/NoUnneededCurlyBracesFixerTest.php]: they don't, I just kept on writing tests :P

Comment (SpacePossum) [inline tests/Fixer/ControlStructure/NoUnneededCurlyBracesFixerTest.php]: fixed :)

File: src/Fixer/ControlStructure/NoUnneededCurlyBracesFixer.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ */
+final class NoUnneededCurlyBracesFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Removes unneeded curly braces that are superfluous and aren\'t of a control structure\'s body.',
```
Comment (SpacePossum) [inline src/Fixer/ControlStructure/NoUnneededCurlyBracesFixer.php]: done :+1:

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**